### PR TITLE
add OAS support to klayout

### PR DESCRIPTION
**Pull request by gadfort**

### CODE REVIEW

File: siliconcompiler/tools/klayout/klayout_export.py [hunk 1]
```diff
@@ -243,15 +257,18 @@ def read_fills(top):
...
 if 'macrolib' in schema.getkeys('asic'):
   libs += schema.get('asic', 'macrolib', step=sc_step, index=sc_index)
 
-in_gds = []
+in_files = []
 for lib in libs:
-  in_gds.extend(schema.get('library', lib, 'output', sc_stackup, 'gds', step=sc_step, index=sc_index))
+  for s in sc_streams:
+    if schema.valid('library', lib, 'output', sc_stackup, s):
+      in_files.extend(schema.get('library', lib, 'output', sc_stackup, s, step=sc_step, index=sc_index))
```
Comment (WRansohoff) [inline siliconcompiler/tools/klayout/klayout_export.py:270]: Same question about duplicate entries in the streams array as above.

**Review by WRansohoff — COMMENTED** [review]
Looks good, just a few minor questions and comments. The main question is, should the arrays containing a list of streams be deduplicated?

File: siliconcompiler/tools/klayout/export.py [hunk 2]
```diff
@@ -23,15 +23,39 @@ def setup(chip):
...
+    if stackup and targetlibs:
         macrolibs = chip.get('asic', 'macrolib', step=step, index=index)
 
         chip.add('tool', tool, 'task', task, 'require', ",".join(['asic', 'logiclib']), step=step, index=index)
         chip.add('tool', tool, 'task', task, 'require', ",".join(['option', 'stackup']), step=step, index=index)
-        chip.add('tool', tool, 'task', task, 'require',  ",".join(['pdk', pdk, 'layermap', 'klayout', 'def','gds', stackup]), step=step, index=index)
+        req_set = False
+        for s in sc_stream_order:
+            if chip.valid('pdk', pdk, 'layermap', 'klayout', 'def', s, stackup):
+                chip.add('tool', tool, 'task', task, 'require', ",".join(['pdk', pdk, 'layermap', 'klayout', 'def', s, stackup]), step=step, index=index)
```
[the remark this answers is not in the document]
Comment (gadfort) [reply siliconcompiler/tools/klayout/export.py:42]: the break on the following line would avoid this.

File: siliconcompiler/tools/klayout/klayout_export.py (hunk 1 above)
Comment (gadfort) [reply siliconcompiler/tools/klayout/klayout_export.py:270]: same, there is a break that would avoid this

File: siliconcompiler/tools/klayout/export.py [hunk 3]
```diff
@@ -23,15 +23,39 @@ def setup(chip):
...
     stackup = chip.get('option', 'stackup')
     pdk = chip.get('option', 'pdk')
-    if bool(stackup) & bool(targetlibs):
+
+    # Set stream extension
+    streams = ('gds', 'oas')
+    chip.set('tool', tool, 'task', task, 'var', 'stream', 'gds', step=step, index=index, clobber=False)
+    chip.set('tool', tool, 'task', task, 'var', 'stream', f'Extension to use for stream generation ({streams})', field='help')
+    stream = chip.get('tool', tool, 'task', task, 'var', 'stream', step=step, index=index)[0]
+    sc_stream_order = [stream, *streams]
```
[the remark this answers is not in the document]
Comment (gadfort) [reply siliconcompiler/tools/klayout/export.py]: cleaned up to avoid the duplication

**Review by WRansohoff — APPROVED** [review]
Looks great!

**Pull request merged**